src/main/java/de/tum/in/www1/artemis/repository/ComplaintRepository.java

```diff
@@ -38,6 +41,17 @@
     @Query("SELECT c FROM Complaint c LEFT JOIN FETCH c.result r LEFT JOIN FETCH r.assessor LEFT JOIN FETCH r.participation p LEFT JOIN FETCH p.exercise e LEFT JOIN FETCH r.submission WHERE e.id = :#{#exerciseId}")
     Optional<List<Complaint>> findByResult_Participation_Exercise_IdWithEagerSubmissionAndEagerAssessor(@Param("exerciseId") Long exerciseId);
 
+    /**
+     * Count the number of unaccepted complaints of a student in a given course. Unaccepted means that they are either open/unhandled or rejected. We use this to limit the number
+     * of complaints for a student in a course.
+     *
+     * @param studentId the id of the student
+     * @param courseId  the id of the course
+     * @return the number of unaccepted
+     */
+    @Query("SELECT count(c) FROM Complaint c  WHERE c.student.id = :#{#studentId} AND c.result.participation.exercise.course.id = :#{#courseId} AND (c.accepted = false OR c.accepted is null)")
```

Comment:
Does this always work? You don't do a `FETCH` :thinking:

Reply:
Yes, I think it should work like this. We do not need a `FETCH` here as all the fields we are referring to are fetched eagerly from the database anyway. Or am I missing something?